### DISCUSSION TITLE
How to change learning rate dynamically during training process?

### DISCUSSION THREAD

**Discussion by MSTK2014** [open-mmlab/mmdetection3d#2072] [General] [answered]
How to change learning rate dynamically during training process? Not set a fixed learning rate in config file, but dynamically change learning rate during training process.

**Comment by JingweiZhang12** [comment, marked as answer]
You can write a hook to set a custom learning rate if the current learning schedule does not meet your need. https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/param_scheduler_hook.py
Or, you can write a new learning schedule. https://github.com/open-mmlab/mmengine/blob/main/mmengine/optim/scheduler/param_scheduler.py

**Reply by MSTK2014** [reply]
Thank you!
